frontend/src/components/ColorLegend.svelte

```diff
@@ -1,28 +1,96 @@
 <script lang="ts">
-    export let colors;
+    export let styles;
+    import G6, { Graph } from "@antv/g6";
+    import { onMount } from "svelte";
+    import { createEventDispatcher } from "svelte";
+
+    const options = {
```

Comment:
The options are given as parameters in the app.svelte currently. Changing this pattern will impact other branches like #4 creating unnecessary conflicts in my opinion. 

(I don't understand why all this has been changed, but I am sure you had a reason! I just wonder if there is a way to have something that is less impacting9